src/ILCompiler.Compiler/src/CppCodeGen/CppWriter.cs

```diff
@@ -729,189 +729,194 @@ private String GetCodeForVirtualMethod(MethodDesc method, int slot)
+                        nodeDataDecl.Append("(),");
+                    }
+                    else
+                    {
+                        // TODO Add support for other relocs
+                        nodeDataDecl.Append("NULL,");
+                    }
+                    relocCounter++;
+                }
+                else
+                {
+                    nodeDataDecl.Append(GetFormattedByteArray(byteData, divisionStartIndex, divisionStartIndex + nodeDataDivs[i].Item2));
+                    nodeDataDecl.Append(",");
+                    divisionStartIndex += nodeDataDivs[i].Item2;
+
+                }
+                nodeDataDecl.AppendLine();
+                nodeDataDecl.Exdent();
+
+
             }
-            else
+            return nodeDataDecl.ToString();
+
+        }
+        private String GetCodeForNodeStruct(List<Tuple<bool, int>> nodeDataDivs, DependencyNode node)
+        {
+            CppGenerationBuffer nodeStructDecl = new CppGenerationBuffer();
+            int relocCounter = 1;
+            int i = 0;
+            nodeStructDecl.Indent();
+
+            for (i = 0; i < nodeDataDivs.Count; i++)
             {
-                var baseType = type.BaseType;
-                if (baseType != null)
+                Tuple<bool, int> div = nodeDataDivs[i];
+                if (div.Item1)
                 {
-                    sb.Append(GetCppMethodDeclarationName(type.BaseType, "__getMethodTable"));
-                    sb.Append("()");
+                    nodeStructDecl.Append("void* reloc");
+                    nodeStructDecl.Append(relocCounter);
+                    nodeStructDecl.Append(";");
+                    relocCounter++;
                 }
                 else
                 {
-                    sb.Append("NULL");
+                    nodeStructDecl.Append("unsigned char data");
+                    nodeStructDecl.Append((i + 1) - relocCounter);
+                    nodeStructDecl.Append("[");
+                    nodeStructDecl.Append(div.Item2);
+                    nodeStructDecl.Append("];");
+
                 }
+                nodeStructDecl.AppendLine();
             }
-            sb.Exdent();
-            sb.AppendLine();
-            sb.Append("},");
+            nodeStructDecl.Exdent();
 
-            // virtual slots
-            if (!type.IsGenericDefinition && ((DependencyNode)_compilation.NodeFactory.ConstructedTypeSymbol(type)).Marked)
-                AppendVirtualSlots(sb, type, type);
-
-            sb.Exdent();
-            sb.AppendLine();
-            sb.Append("};");
-            sb.AppendLine();
-            sb.Append("return (MethodTable *)&mt.EEType;");
-            sb.Exdent();
-            sb.AppendLine();
+            return nodeStructDecl.ToString();
+        }
+        private String GetFormattedByteArray(byte[] array, int startIndex, int endIndex)
+        {
+            CppGenerationBuffer sb = new CppGenerationBuffer();
```

Comment:
(Once you do that, it may look better to not even have it in separate method and just inline it in one place where it is used.)